src/toc.tsx

```diff
@@ -198,6 +232,9 @@ export interface ITOCTreeProps extends React.Props<TOCTree> {
    * A list of IHeadings to render.
    */
   toc: IHeading[];
+  widget: TableOfContents;
+  contextChanged: boolean;
```

Comment:
I think of react props as being stateless. Having a `contextChanged` boolean to me connotes state. What is the purpose of this, can can we think of another way to do it? It also makes me nervous to pass the direct parent of the `TOCTree` in as a prop. A more idiomatic way would be for the `TableOfContents` to pass in some sort of callback to accomplish what it wants.